configdialog.cpp

```diff
@@ -518,6 +535,271 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
 	}
 }
 
+std::wstring PaletteCustomizeDialog::export_palettes() {
+	auto all_palettes = m_palette_repo.get_all_custom_palettes();
+	std::wstring palettes_string = L"";
+
+	for (const auto &palette : all_palettes) {
+		auto name = palette.name;
+		auto *data = palette.data;
+
+		palettes_string.append(palette.name + L": ");
+		std::wstring colors = m_palette_repo.serialize(*data) + L";";
+		// Add extra "fake" colors so that a palette definition is guaranteed to be longer than the maximum allowed name length
+		while (colors.length() < PaletteCustomizeDialog::MaxPaletteNameSize) {
+			colors.append(L"#000000;");
+		}
+		palettes_string.append(colors);
+		palettes_string.append(L"\r\n");
+	}
+	palettes_string.append(L"~");
+
+	return palettes_string;
+}
+
+void PaletteCustomizeDialog::import_palettes(std::wstring *palettes) {
```

Comment:
This function is over 200 lines long, with lines that have up to seven levels of indentation. Frankly, that is ludicrous. I understand that a lot of complexity comes from doing the in-depth error reporting, but I am worried that even then this has been over-engineered. It is a very easy mistake to make, because programming and solving problems and being clever is fun, but it is a mistake nonetheless. We are left with an enormous, nearly untouchable bird's nest that I don't even want to try to understand. Am I going to be able to come into and fix bugs if I need to? Are you going to be able to come back to this and change it when our format changes 6 months later?

Now I can tell a lot of love and care went into this, so it probably works fine, and the last thing I want you to do after solving this hard problem is tell you to go throw it all out. But this is one of the reasons I like to not solve hard problems - it would have been much easier and more graceful and maintainable without detailed error reporting, it might have been just a few calls to `split` and few checks for vector size to do basic sanity checks. And seriously, what errors are we _really_ expecting from copying and pasting some text?

Again, at this point all I care about is that it works, so my "real" review of this once it is closer to done will not really be a line-by-line thing but a suite of runtime tests. Please don't waste a bunch of time simplifying it now, just keep going with actual importing and we'll come back to this if we find out we need to.